app/flatpak-builtins-build-init.c

```diff
@@ -390,8 +397,30 @@ flatpak_builtin_build_init (int argc, char **argv, GCancellable *cancellable, GE
                               optional_extension_tag);
     }
 
+  /* Do the rest of the work as a keyfile, as we need things like full escaping, etc.
+   * We should probably do everything this way actually...   */
+  if (!g_key_file_load_from_data (keyfile, metadata_contents->str, metadata_contents->len, 0, NULL))
+    return flatpak_fail (error, "Internal error parsing generated keyfile");
+
+  for (i = 0; opt_extensions != NULL && opt_extensions[i] != NULL; i++)
+    {
+      g_auto(GStrv) elements = NULL;
+      g_autofree char *groupname = NULL;
+
+      elements = g_strsplit (opt_extensions[i], "=", 3);
+      if (g_strv_length (elements) < 2)
+        return flatpak_fail (error, _("Too few elements in --extension argument %s, format should be NAME=VAR[=VALUE]"), opt_extensions[i]);
+
+      groupname = g_strconcat (FLATPAK_METADATA_GROUP_PREFIX_EXTENSION,
+                               elements[0], NULL);
+
+      g_key_file_set_string (keyfile, groupname, elements[1], elements[2] ? elements[2] : "true");
```

Comment:
Not sure I understand this part. If `elements[2]` is `NULL` I am assuming that this means that the user specified `--extension NAME`. Doesn't that contract the `flatpak_fail` above?